### PR TITLE
Added new cl_khr_semaphore tests to verify clCreateSemaphoreWithPropertiesKHR negative results

### PR DESCRIPTION
**Pull request by shajder**
According to work plan from #1691, new clSemaphoreWrapper introduced to avoid duplication of code

### CODE REVIEW

File: test_conformance/extensions/cl_khr_semaphore/test_semaphores_negative_create.cpp [hunk 1]
```diff
@@ -0,0 +1,672 @@
...
+            (cl_semaphore_properties_khr)CL_SEMAPHORE_HANDLE_OPAQUE_FD_KHR,
+            (cl_semaphore_properties_khr)
+                CL_SEMAPHORE_EXPORT_HANDLE_TYPES_LIST_END_KHR,
+            0
+        };
+
+        semaphore_second =
+            clCreateSemaphoreWithPropertiesKHR(context, sema_2_props, &err);
+        test_failure_error(
+            err, CL_INVALID_DEVICE,
```
Comment (saurabhnv) [inline test_conformance/extensions/cl_khr_semaphore/test_semaphores_negative_create.cpp:593]: Shouldn't this be **CL_INVALID_OPERATION**?

Comment (shajder) [reply test_conformance/extensions/cl_khr_semaphore/test_semaphores_negative_create.cpp]: Corrected, thanks!

**Review by joshqti — COMMENTED** [review]
CL_INVALID_SEMAPHORE_KHR is an error code.  Define an invalid property local to the test, this is not correct usage of the macro

**Comment by shajder** [comment]
> CL_INVALID_SEMAPHORE_KHR is an error code. Define an invalid property local to the test, this is not correct usage of the macro

@joshqti thanks, do you have other values on your mind ? Please keep in mind it supposed to be incorrect usage because this is negative test! `CL_INVALID_SEMAPHORE_KHR` is neither valid name nor property in this case.

File: test_conformance/extensions/cl_khr_semaphore/test_semaphores_negative_create.cpp [hunk 2]
```diff
@@ -0,0 +1,672 @@
...
+            cl_semaphore_properties_khr sema_props[] = {
+                (cl_semaphore_properties_khr)CL_INVALID_SEMAPHORE_KHR,
+                (cl_semaphore_properties_khr)CL_SEMAPHORE_TYPE_BINARY_KHR, 0
+            };
+
+            semaphore =
+                clCreateSemaphoreWithPropertiesKHR(context, sema_props, &err);
+            test_failure_error(
+                err, CL_INVALID_PROPERTY,
+                "Unexpected clCreateSemaphoreWithPropertiesKHR return");
```
Comment (bashbaug) [inline test_conformance/extensions/cl_khr_semaphore/test_semaphores_negative_create.cpp:99]: I think there are two possible errors here:

1. There is an invalid property, which would return `CL_INVALID_PROPERTY`.
2. The properties  do not contain the required properties to create a semaphore (specifically, the semaphore type), which would return `CL_INVALID_VALUE`.

We should either allow for both error conditions, or we should disambiguate the errors so only `CL_INVALID_PROPERTY` is returned.

**Review by paulfradgley — DISMISSED** [review]
Looks good to me.

**Review by ahesham-arm — APPROVED** [review]

**Comment by kpet** [comment]
2024/07/02 teleconference: @shajder Could you rebase/resolve conflicts please?

**Comment by nikhiljnv** [comment]
Summarizing discussion from memory subgroup call of July 9th 2024.

Per Joshua's comment, using CL_INVALID_SEMAPHORE error code as a property value is confusing.
Ben suggested two options

-     Clarify in a comment that CL_INVALID_SEMAPHORE which is an error code is actually an invalid value that can be passed to test error      condition.
-     Alternatively, pass a different value instead of CL_INVALID_SEMAPHORE to avoid the confusion.

**Comment by shajder** [comment]
> Per Joshua's comment, using CL_INVALID_SEMAPHORE error code as a property value is confusing. Ben suggested two options

corrected according to mentioned suggestion

**Review by nikhiljnv — APPROVED** [review]
LGTM.

**Comment by nikhiljnv** [comment]
Merging as discussed in memory subgroup call on August 6th 2024.

**Pull request merged**